### DISCUSSION TITLE
Network timeout at http://192.168.1.16:1400/MediaRenderer/AVTransport/Control

### DISCUSSION THREAD

**Discussion by Lbjerring** [hklages/node-red-contrib-sonos-plus#178] [Q&A] [answered]
Hi

Thank you for at really nice addin to Node red, I like it a lot.

However, I have a problem that I need some help to troubleshoot. I created a flow that starts a radio stream every morning on my Sonos. But very often I get a network timeout when the Sonos has been idle all night. If I try to start the flow a few times it will start playing.

I'm able to ping Sonos and I can access http://192.168.1.16:1400/MediaRenderer/AVTransport/Control from a browser. I can also run "group.get.state" and see the details. So I'm pretty sure the Sonos is responding to network traffic. The Sonos is connected to wifi (unifi). Node red is running on a Home Assistant VM which is connected to LAN (by cable). 

So I don't understand why I often get the timeout when starting "group.play.tunein". Hope someone can point me in the right direction.

`"group.play.tunein:network timeout at: http://192.168.1.16:1400/MediaRenderer/AVTransport/Control :: Details: {"message":"network timeout at: http://192.168.1.16:1400/MediaRenderer/AVTransport/Control","type":"request-timeout","stack":"FetchError: network timeout at: http://192.168.1.16:1400/MediaRenderer/AVTransport/Control\n    at Timeout.<anonymous> (/config/node-red/node_modules/node-fetch/lib/index.js:1476:13)\n    at listOnTimeout (internal/timers.js:557:17)\n    at processTimers (internal/timers.js:500:7)"}"`

**Comment by hklages** [comment, marked as answer]
Hi.
Maybe setting the stream takes to long (buffering) and then the timeout
triggers the error message and the following play command is not being
executed.

Are you able to activate debugging on your system? What system?

**Reply by Lbjerring** [reply]
Hi, 
That sounds very likely. I'm pretty new to Home Assistant and Node Red. So I'm not sure how to activate debugging. Can you tell me where to activate it and how to get the log's?

As the problem only occurs after hours of standby. I might take a day or two to have useful debug.

**Reply by Lbjerring** [reply]
Hi,
I tried to create a script in HomeAssistant that start the stream and then call this script from my NodeRed flow. That seems to work well without timeouts. So it does seem like a timeout problem when the stream is called from the NodeRed extension. 

But so far I got the result I needed. No need to spent more time troubleshooting this :-)
Thanks.

**Comment by hklages** [comment]
Great. Anyway I have to find a solution for that case and will work on it December, when I am back at home.

**Comment by ericbogs** [comment]
Hi, I'm triggering this network timeout as well, on group.play.export of a Spotify playlist with 2200 songs. Is there a way to configure the network timeout to be longer than 5000ms?

Use case:

Hubitat triggers Node-RED to get a random favorite from Sonos config (mysonos.export.item) and then play the favorite (group.play.export).

Most of my favorites are either stations or playlists with a few hundred items, and can play in under 5000ms. Some of my favorites are playlists with 2000+ items, which is causing a network timeout error. 

Is there a way to configure the network timeout to be longer than 5000ms?

This is the offending Spotify playlist:

https://open.spotify.com/playlist/69SNbRN7bKcPiFXxUpT4la?si=38419f2c50a448c8

I also tried switching to group.play.mysonos (instead of mysonos.export.item & group.play.export) with the same result.

Thanks!

**Comment by hklages** [comment]
Hi Eric. 

This package is based on node-ts. I just received the information that a new beta is available with time out 30 seconds. 
That should be enough. 

I need a few days to update/test and publish - most likely in 2022-01-0x it will be available.

**Reply by ericbogs** [reply]
Wonderful, thanks!

**Comment by hklages** [comment]
I just published 5.2.8  - please check. 

Does that new release solve the timeout issue? Please provide feedback.

**Comment by ericbogs** [comment]
Yup, upgrading to 5.2.8 fixed it. Many thanks!

**Comment by hklages** [comment]
Thanks for your feedback!
